src/marvin/utilities/llms.py

```diff
@@ -139,3 +146,54 @@ def prepare_messages(
         else:
             raise ValueError(f"Unrecognized role: {msg.role}")
     return langchain_messages
+
+
+def message_sort_key(
```

Comment:
I think this can be simplified a lot and remove this sort key, which is a bit confusing. 

I suggest this pseudocode:

```python
keep_indices = [...] # the index of any system message, we must keep those
count = sum(token_count(messages[i]) for i in keep_indices)

# next keep as many messages as possible in reverse order
for i, msg in reversed(enumerate(messages)):
    if i not in keep_indices:
	if count + token_count(msg) <= max_count:
            keep_indices.append(i)

final_messages = [messages[i] for i in sorted(keep_indices)]    
```

Note here I'm counting system messages as the ones we intentionally send, not plugin responses. Plugin resposnes can be counted like any other message in reverse chronological order.